### ISSUE TITLE
Consider moving from js-yaml to yaml module

### ISSUE THREAD

**Issue by MikeRalphson** [wework/speccy#243]
## Detailed description

`js-yaml` has some bugs related to the YAML 1.2 specification, in that it appears to convert strings which contain only digits and begin with a 0 into octal integers when it should not. This *may* only be the case where the `{json:true}` option has been provided. It also seems to drop seconds/fractions of a second from date-time string properties.

The [yaml](https://github.com/eemeli/yaml) module is tested against the official `yaml-test-suite` and is smaller (not including the `esprima` engine). It has far fewer open issues (but may be much less utilised in the wild so far). It also has the ability to build an abstract-syntax-tree or concrete-syntax-tree built in. It will also hopefully be gaining some nice features such as:

* lookups of errors to **line number** and column positions https://github.com/eemeli/yaml/pull/68
* tagging of the syntax-tree with JSON pointers for error lookups
* optional ability to preserve comments when converting to JS objects

(**if** my PRs are accepted).

## Context

`oas-kit` is going to migrate so `node_modules` de-duplication will result in an even smaller installed size. This will probably be a breaking change in `oas-kit` (all version numbers bumped by a major revision).

## Possible implementation

Replace `js-yaml` with `yaml`, `safeLoad` with `parse`, `safeDump` with `stringify`.

## Your environment

all

## More...

This issue is a heads-up really. But I'm happy to PR the changes.

**Comment by dlouzan** [comment]
@philsturgeon Today I found I'm experiencing some issues when using response types that include `+` and `{}` characters, when resolving to external references.

Example:
```yaml
  /foo:
    get:
      summary: Foo
      operationId: foo
      responses:
        200:
          $ref: "./responses.yaml#/components/responses/Foo"
```

```yaml
components:
  responses:
    Foo:
      description: Success, returning the root context and list of things
      content:
        application/ld+json:
          schema:
            $ref: schemas.yaml#/components/schemas/Foo
```

When I try to `speccy resolve` this generates an incorrect yaml file such as below with two equal keys, but one key missing the `+` (I don't think that's even valid yaml). Also JSON pointers don't seem to work properly when referencing anything including these special characters in the path.

```yaml
      responses:
        '200':
          description: 'Success, returning the root context and list of things'
          content:
            application/ld+json:
              schema:
                $ref: '#/paths/~1foo/get/responses/200/content/application~1ld%2Bjson/schema'
                x-miro: 'schemas.yaml#/components/schemas/Foo'
            application/ld json:
              schema:
                title: Thing Description graph
                description: A Graph of Things in JSON-LD format.
                type: object
...
```

Might this be related or should I open a new issue?

**Comment by MikeRalphson** [comment]
@dlouzan sounds very similar to a new issue #249 - maybe we can move the discussion there?

**Comment by philsturgeon** [comment]
I'll do this when #250 is finished (adding better testing to the command files).

**Comment by pderaaij** [comment]
@MikeRalphson yoou mentioned working on the yaml module, but you meant the yaml package right? Not changing `speccy` to make use of `yaml`.

**Comment by pderaaij** [comment]
Okay, few days became this hour ;-)

**Comment by dlouzan** [comment]
I lost track of the status of this, is it now possible to get proper line numbers on errors when validating yaml? Today I found some issues with yaml comments, and it seems the latest master doesn't still include this functionality.

**Comment by eemeli** [comment]
Line/column locations should be available in `yaml` at `error.source.rangeAsLinePos`, starting from its version 1.3.

**Comment by dlouzan** [comment]
Ok, I see that #267 is still open, and uses 1.2.

@philsturgeon @MikeRalphson How easy would be to integrate 1.3 *and* modify the error reporting to get meaningful reports? I'm not familiar with that part of the code (I'm not even sure if this would need to be modified in oas-kit).

**Comment by djtarazona** [comment]
This has been addressed in [v0.10.0](https://github.com/wework/speccy/releases/tag/v0.10.0)
